### PR TITLE
CPDTP-329 Add in model validations for participant_declarations

### PR DESCRIPTION
**Pull request by blackrat**
### Context

Now that this is using STI we can put in proper course type validations for NPQ and ECF courses.

### Changes proposed in this pull request

Inclusion validations to make sure that the correct course identifier is used for each declaration type. For NPQ it reads from the NPQCourse identifiers held in the database, so that if more courses are added, we can validate against those without having to make code changes too.
Presence validation to ensure that the required fields are passed to the model.

Factories and tests updated to ensure that correct course_identifiers are linked to the correct profile types.

### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All pages have visual tests via cypress + percy
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### CODE REVIEW

File: spec/models/participant_declaration_spec.rb [hunk 1]
```diff
@@ -16,11 +16,14 @@
...
                cpd_lead_provider: call_off_contract.lead_provider.cpd_lead_provider,
-               profile_type: profile_type)
+               profile_type: profile_type,
+               course_identifier: course_identifier)
       end
 
       context "for mentor was created" do
         let(:profile_type) { :mentor_profile }
+        let(:course_identifier) { "ecf-mentor" }
+        let(:course) { :ecf_mentor }
```
Comment (kicferk1) [inline spec/models/participant_declaration_spec.rb:26]: Is `course` used? Doesn't look like it.

File: spec/models/participant_declaration_spec.rb [hunk 2]
```diff
@@ -29,6 +32,7 @@
 
       context "for early career teacher was created" do
         let(:profile_type) { :early_career_teacher_profile }
+        let(:course_identifier) { "ecf-induction" }
```
Comment (kicferk1) [inline spec/models/participant_declaration_spec.rb:35]: Same with `course_identifier` actually

Comment (blackrat) [reply spec/models/participant_declaration_spec.rb:35]: > Same with `course_identifier` actually

`course_identifier` is used in the create factory (line 20)

File: spec/models/participant_declaration_spec.rb (hunk 1 above)
Comment (blackrat) [reply spec/models/participant_declaration_spec.rb:26]: Don't think it is in this one. I'll pull it.

**Review by kicferk1 — APPROVED** [review]

File: app/models/participant_declaration.rb [hunk 3]
```diff
@@ -5,6 +5,9 @@ class ParticipantDeclaration < ApplicationRecord
   belongs_to :cpd_lead_provider
   belongs_to :user
 
+  validates :course_identifier, inclusion: { in: :valid_courses }
```
Comment (ethanmills) [inline app/models/participant_declaration.rb:8]: Do we need an `allow_blank: true` here to prevent duplicates?

**Pull request merged**